Source/UZKArchive.m

```diff
@@ -419,66 +419,78 @@ - (BOOL)extractFilesTo:(NSString *)destinationDirectory
                     if (info.isDirectory) {
                         continue;
                     }
-                    
-                    NSData *data = [self readFile:info.filename
-                                           length:info.uncompressedSize
-                                            error:&strongError];
-                    
-                    if (!data && strongError) {
-                        [self assignError:&strongError code:UZKErrorCodeFileRead
-                                   detail:[NSString localizedStringWithFormat:NSLocalizedString(@"Error reading current file '%@'", @"Detailed error string"),
-                                           info.filename]];
+                    					
+					BOOL isDirectory = YES;
+					NSString *extractDir = extractPath.stringByDeletingLastPathComponent;
+					if (![fm fileExistsAtPath:extractDir]) {
+						BOOL directoriesCreated = [fm createDirectoryAtPath:extractDir
+												withIntermediateDirectories:YES
+																 attributes:nil
+																	  error:error];
+						if (!directoriesCreated) {
+							[self assignError:&strongError code:UZKErrorCodeOutputError
+									   detail:[NSString localizedStringWithFormat:NSLocalizedString(@"Failed to create destination directory: %@", @"Detailed error string"),
+											   extractDir]];
+							return;
+						}
+					} else if (!isDirectory) {
+						[self assignError:&strongError code:UZKErrorCodeOutputErrorPathIsAFile
+								   detail:[NSString localizedStringWithFormat:NSLocalizedString(@"Extract path exists, but is not a directory: %@", @"Detailed error string"),
+										   extractDir]];
+						return;
+					}
+
+					
+					NSURL *deflatedDirectoryURL = [NSURL fileURLWithPath:destinationDirectory];
+					NSURL *deflatedFileURL = [deflatedDirectoryURL URLByAppendingPathComponent:info.filename];
+					NSString *path = deflatedFileURL.path;
+					
+					BOOL createSuccess = [fm createFileAtPath:path
+													 contents:nil
+												   attributes:nil];
+
+					if (!createSuccess) {
+						[self assignError:&strongError code:UZKErrorCodeOutputError
+								   detail:[NSString localizedStringWithFormat:NSLocalizedString(@"Error creating current file (%d) '%@'", @"Detailed error string"),
+										   strongError, info.filename]];
+						return;
+					}
+										
+					NSFileHandle *deflatedFileHandle = [NSFileHandle fileHandleForWritingToURL:deflatedFileURL
+                                                                                         error:&strongError];
+
+					
+                    if (!deflatedFileHandle) {
+                        [self assignError:&strongError code:UZKErrorCodeFileHandleCreate
```

Comment:
I don't think it's necessary to create a new error code for this case. Just use the `UZKErrorCodeOutputError` code used elsewhere in the method.